sympy/simplify/_cse_diff.py

```diff
@@ -0,0 +1,255 @@
+"""Module for differentiation using CSE."""
+
+from sympy import cse, Matrix, Derivative, MatrixBase
+from sympy.utilities.iterables import iterable
+
+
+def _process_cse(replacements, reduced_expressions):
```

Comment:
```suggestion
def _remove_cse_from_derivative(replacements, reduced_expressions):
```

that could be a more informative name. "process" can mean anything